context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx

```diff
@@ -45,11 +52,33 @@ function Visualization(props) {
     vizTheme,
     vizEscSnackbarIsOpen,
     setVizEscSnackbarIsOpen,
+    setVitessceConfig,
+    onCopyUrlWarning,
+    onCopyUrlSnackbarOpen,
+    setOnCopyUrlSnackbarOpen,
   } = useVisualizationStore(visualizationStoreSelector);
 
-  const [vitessceErrors, setVitessceErrors] = useState([]);
-  const [vitessceSelection, setVitessceSelection] = useState(0);
+  // Get the vitessce configuration from the url if available and set the initial selection if it is a multi-dataset.
+  const [initializedVitData, initialSelection] = useMemo(() => {
+    const queryString = window.location.href.split('#')[1];
+    const vitessceURLConf = queryString?.length > 0 ? decodeURLParamsToConf(queryString) : null;
+    const initialSelectionFromUrl =
+      Array.isArray(vitData) && Math.max(0, vitData.map(({ name }) => name).indexOf(vitessceURLConf?.name));
+    let initializedVitDataFromUrl = vitData;
+    if (Array.isArray(vitData)) {
+      initializedVitDataFromUrl[initialSelectionFromUrl] = vitessceURLConf || vitData[initialSelectionFromUrl];
+    } else {
+      initializedVitDataFromUrl = vitessceURLConf || vitData;
+    }
+    setVitessceConfig(initializedVitDataFromUrl);
+    return [initializedVitDataFromUrl, initialSelectionFromUrl];
+  }, [vitData, setVitessceConfig]);
 
+  const [vitessceErrors, setVitessceErrors] = useState([]);
+  const [vitessceSelection, setVitessceSelection] = useState(initialSelection);
+  const handleVitessceConfigDebounced = useCallback(debounce(setVitessceConfig, 250, { trailing: true }), [
```

Comment:
Was use callback needed here? Is it because of the vitessce internals?

Reply:
From standup: Does it make sense to move the useCallback into vitessce? Possibly, but probably fine as is.

Reply:
Not needed (or rather, not performant).  Has been removed.